### ISSUE TITLE
[1.2.0] Do not import host_snmp_cache 

### ISSUE THREAD

**Issue by jackouille-ch** [Cacti/cacti#1754]
Some install have over the 300'000 entries maximum defined and because it's only cache value, it would be great to no import them into Cacti 1.2.0

**Comment by cigamit** [comment]
I have no idea what you are asking about here.  The host_snmp_cache is critical to graph operations.  So, it's not going anywhere.

**Comment by jackouille-ch** [comment]
When doing an import, table host_snmp_cache can be very huge (mybe it's the biggest table).
When we perform an import, the install script is limiting the entries to 300'000 and this table also generate a problem.
When I was importing my database (with MySQLAdmin), I was obliged to flush it's content before being able to continue install.
So I opened this ticket in order to not import this table within install process. This table might regenerated automatically while in production and when you are doing an import, it means that you have planed a downtime or database cleaning.

**Comment by jackouille-ch** [comment]
Here is what I got today:
![image](https://user-images.githubusercontent.com/35721406/43309259-997cfc4c-9184-11e8-9ad3-73fffd72840f.png)
![image](https://user-images.githubusercontent.com/35721406/43309287-ab4d09f8-9184-11e8-8613-e716e01e37b9.png)

But I am able to continue without flushing the table. So it's not blocking, simply generating a Warning.
I opened this ticket in order to simply not import this table if it's over 300'000 entries, so not generating the Warning (because we no other choice that continue).

**Comment by netniV** [comment]
The installer does block for certain circumstances were you have large tables that it would be unable to convert without causing potential timeouts.  Part of that was done before the background installer was implemented so in theory we should be able to continue.  Changing from MyISAM to InnoDB isn't as important as ensure the collation matches so we could make it more a 'suggestion' but I'll discuss this with @cigamit later to decide what we do.

**Comment by cigamit** [comment]
We make the change in tables optional.  We also need to ensure that we give you the option to keep some of the engines in MyISAM.  We will keep this open as a result.

**Comment by cigamit** [comment]
Oh, that is odd.  I just looked at the error message.  I did notice that feature of the upgrade script.  Well, we are working on it.

**Comment by netniV** [comment]
That is because the table is in latin1, failing to convert the table to utf8 will cause collation issues with other tables.  If the table is MyISAM instead of InnoDB, that can be allowed since the storage engine does not affect data queries or insert/updates.

**Comment by netniV** [comment]
I would agree engine shouldn't be required, just optional.  UTF8 will cause errors and it should be required.  I like the idea of a skip table option though it should stop if an invalid table name is specified.

**Comment by cigamit** [comment]
I've increased the threshold to 1E06 for skipping conversion and added the option to skip innodb conversion selectively.

**Comment by jackouille-ch** [comment]
Ok so for me you can close this ticket, thanks.
